### PR TITLE
Report exit codes for all relevant sub-commands

### PR DESCRIPTION
**Pull request by githubert**
*Before submission, please check that ...*

- [ ] ~the code follows the [Clean Code](https://clean-code-developer.com/) guidelines.~
- [ ] ~the necessary tests are either created or updated.~
- [ ] all status checks (Travis CI, SonarCloud, etc.) pass.
- [x] your commit history is cleaned up.
- [x] you updated the changelog.
- [ ] ~you updated necessary documentation within [retest/docs](https://github.com/retest/docs).~

## Description

The sub-commands should now use exit codes that are hopefully more in line with what a user might expect.

### State of this PR

All done.

### CODE REVIEW

File: src/main/java/de/retest/recheck/cli/subcommands/Completion.java [hunk 1]
```diff
@@ -30,9 +36,13 @@ public void run() {
 		final String script = AutoComplete.bash( root.getCommandName(), root );
 		if ( script.isEmpty() ) {
 			logger.error( "Failed to generate the auto completion script." );
+			exitCode = SOFTWARE;
 		} else {
 			logger.info( script );
 		}
 	}
 
+	@Override public int getExitCode() {
```
Comment (lemoncurry) [inline src/main/java/de/retest/recheck/cli/subcommands/Completion.java]: Here the formatting is a little bit off. Annotation should be above the method.

File: src/main/java/de/retest/recheck/cli/subcommands/Ignore.java [hunk 2]
```diff
@@ -84,10 +92,16 @@ public void run() {
 				saveRecheckIgnore();
 			} catch ( final Exception e ) {
 				ErrorHandler.handle( e );
+				exitCode = SOFTWARE;
```
Comment (lemoncurry) [inline src/main/java/de/retest/recheck/cli/subcommands/Ignore.java]: Maybe put before the ErrorHandler so the code is changed in any case.

Comment (lemoncurry) [reply src/main/java/de/retest/recheck/cli/subcommands/Ignore.java]: Also in all other cases.

File: src/main/java/de/retest/recheck/cli/subcommands/account/Show.java [hunk 3]
```diff
@@ -19,7 +24,11 @@ public void run() {
 			logger.info( "This is your API key: {}", apiKey );
 		} else {
 			logger.info( "No API key found." );
+			exitCode = SOFTWARE;
 		}
 	}
 
+	@Override public int getExitCode() {
```
Comment (lemoncurry) [inline src/main/java/de/retest/recheck/cli/subcommands/account/Show.java]: Small formatting problem. Method should be below the annotation.

File: src/main/java/de/retest/recheck/cli/subcommands/Ignore.java (hunk 2 above)
Comment (githubert) [reply src/main/java/de/retest/recheck/cli/subcommands/Ignore.java]: Good catch! I changed it.

**Pull request merged**